starlette/_compat.py

```diff
@@ -0,0 +1,22 @@
+import hashlib
+
+# TODO: Remove when dropping support for PY38. a try/except is used to
+# detect whether the usedforsecurity argument is available as this fix may also
+# have been applied by downstream package maintainers to other versions in
+# their repositories.
```

Comment:
I think this pull request is just about right, yup! 👍

I think we could probably tweak this comment to be a bit more informative. Perhaps something like this?...

```python
# Compat wrapper to always include the `usedforsecurity=...` flag, which is only added from Python 3.9 onwards.
# We use this flag to indicate that we use `md5` hashes only for non-security cases (our ETag checksums).
# If we don't indicate that we're using MD5 for non-security related reasons, then attempting to use
# this function will raise an error when used environments which enable a strict "FIPs mode".
#
# See issue: https://github.com/encode/starlette/issues/1365
```

Reply:
incorporated, thanks!